server.go

```diff
@@ -295,26 +303,76 @@ type Server struct {
 	DecorateReader DecorateReader
 	// DecorateWriter is optional, allows customization of the process that writes raw DNS messages.
 	DecorateWriter DecorateWriter
-
+	// UDP packet or TCP connection queue
+	queue chan *response
+	// Workers count
+	workersCount int32
 	// Shutdown handling
 	lock    sync.RWMutex
 	started bool
 }
 
+func (srv *Server) worker(w *response) {
+	workersCount := atomic.LoadInt32(&srv.workersCount)
+	if workersCount > maxWorkersCount {
+		w.Close()
+		return
+	}
+	atomic.AddInt32(&srv.workersCount, 1)
```

Comment:
It’s not hugely important, but this has a race with the `LoadInt32` above.

This should be written something like:
```go
for {
  count := atomic.LoadInt32(...)
  if count > ... {
    ...
    return
  }
  if atomic.CompareAndSwapInt32(..., count, count + 1) {
    break
  }
}
```

Reply:
Done